bluepyemodel/tools/utils.py

```diff
@@ -256,10 +256,56 @@ def get_amplitude_from_feature_key(feat_key):
     Args:
         feat_key (str): feature key, e.g. IV_40.soma.maximum_voltage_from_voltagebase
     """
-    n = feat_key.split(".")
-    # case where protocol has '.' in its name, e.g. IV_40.0
-    if len(n) == 4 and n[1].isdigit():
-        n = [".".join(n[:2]), n[2], n[3]]
-    protocol_name = n[0]
+    protocol_name = get_protocol_name(feat_key)
 
     return float(protocol_name.split("_")[-1])
+
+
+def combine_parts_if_dot_in_protocol(feature_name):
```

Comment:
Could you say for this function and get_protocol_name and get_loc_name and get_curr_name in their docstrings that it works both with a feature str (e.g. "IV_40.0.soma.v.voltage_base") and for a response key (e.g.""IV_40.0.soma.v"). Also adding examples of expected names would be a +